common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/AuthorizationResponse.java

```diff
@@ -50,6 +50,24 @@
      */
     private String mState;
 
+    /**
+     * Constructor of {@link AuthorizationResponse}.
+     * @param code  The authorization code generated by the authorization server.
+     */
+    public AuthorizationResponse(final String code){
+        new AuthorizationResponse(code, null);
```

Comment:
Can you spot the bug in this code? 😉

Reply:
Helpful hint: try declaring the fields `mCode` and `mState` as final and seeing what happens

Reply:
Such an obvious bug, creating a separate instance which won't be used at all and the variables in intended class will not be initialized.  Sorry to have missing this 😞 . Thank you for spotting this @iambmelt. Fixed it